Add docs for Kafka output UI for Fleet managed agents

kilfoyle opened the pull request:
This adds docs for the new [Kafka output settings UI](https://github.com/elastic/kibana/pull/160112)

 - Setting descriptions came mainly from the [requirements definitions doc](https://docs.google.com/document/d/1SIyfL8lhdeUMsrm0gFHI0Uc8LhNV0aynOYz05O6ZYb8/edit?pli=1) and the [Filebeat Kafka output docs](https://www.elastic.co/guide/en/beats/filebeat/8.9/kafka-output.html).
 - Kafka output is marked as in technical preview.
 - I split the `Elasticsearch`, `Logstash`, and `Kafka` output settings sections into separate sub-pages.
    ![Screenshot 2023-08-02 at 1 13 56 PM](https://github.com/elastic/ingest-docs/assets/41695641/abe3ed59-748e-4273-8a26-3b38ac90af8a)

**[Preview site](https://ingest-docs_336.docs-preview.app.elstc.co/guide/en/fleet/master/kafka-output-settings.html)**

Closes: #68

kilfoyle commented:
Removed the Channel Buffer Size parameter per https://github.com/elastic/kibana/pull/162875#issuecomment-1664323420

nimarezainia commented on docs/en/ingest-management/fleet/fleet-settings.asciidoc:
i belive in the latest iteration "None" is also a valid option  that is - no authentication required. Also Kerberos will be a follow on enhancement.

criamico commented:
Hi @kilfoyle, I think it would be good to involve @szwarckonrad in this review as he actually worked on implementing the feature so he has more in-depth knowledge of the kafka options. I'll request his review as well.

kilfoyle commented:
@faec Would you mind taking a look at these descriptions of the Kafka [authentication settings](https://ingest-docs_336.docs-preview.app.elstc.co/guide/en/fleet/master/kafka-output-settings.html#_authentication_settings), just to be sure I'm on the right track?

juliaElastic reviewed (approved):
LGTM overall, haven't checked the Kafka settings in detail.
It would be nice to add some screenshots about where to find the `Add output` button in `Fleet settings`, and how the `Add new output` flyout looks like. Though the instructions are there as text, so that's fine too.

szwarckonrad commented on docs/en/ingest-management/fleet/fleet-settings-output-kafka.asciidoc:
```diff
@@ -0,0 +1,367 @@
+**Kafka version**
+
+| The Kafka protocol version that {agent} will request when connecting. 
+Defaults to 1.0.0.
```
Perhaps it's worth noting that we provide support for versions ranging from 0.8.2.0 to 2.6.0. However, it's important to highlight that the latest Kafka version (3.x.x) is expected to be compatible when using the version 2.6.0 setting. This is my understanding of the discussion on this matter; perhaps @faec can confirm.

kilfoyle replied:
Added
>Currently Kafka versions from `0.8.2.0` to `2.6.0` are supported, however the latest 
>Kafka version (`3.x.x`) is expected to be compatible when version `2.6.0` is selected.

szwarckonrad commented on docs/en/ingest-management/fleet/fleet-settings-output-kafka.asciidoc:
```diff
@@ -0,0 +1,367 @@
+|
+**Server SSL certificate authorities**
+
+| The CA certificate to use to connect to Kafka. This is the CA used to generate the certificate and key for Kafka. Copy and paste in the full contents for the CA certificate.
+
+This setting is optional.
+
+Click **Add row** to specify additional certificate authories.
+
+// ============================================================================
+
+|
+**Verification mode**
+// from beats/libbeat/docs/shared-ssl-config.asciidoc
+
+| Controls the verification of server certificates. Valid values are:
+
+`Full`::
+Verifies that the provided certificate is signed by a trusted
+authority (CA) and also verifies that the server's hostname (or IP address)
+matches the names identified within the certificate.
+
+`None`::
+Performs _no verification_ of the server's certificate. This
+mode disables many of the security benefits of SSL/TLS and should only be used
+after cautious consideration. It is primarily intended as a temporary
+diagnostic mechanism when attempting to resolve TLS errors; its use in
+production environments is strongly discouraged.
+
+`Strict`::
+Verifies that the provided certificate is signed by a trusted
+authority (CA) and also verifies that the server's hostname (or IP address)
+matches the names identified within the certificate. If the Subject Alternative
+Name is empty, it returns an error.
+
+`Certificate`::
+Verifies that the provided certificate is signed by a
+trusted authority (CA), but does not perform any hostname verification.
+
+The default value is `Full`.
```
We might consider reiterating in both these sections that users who select None/Plaintext authentication will not have access to these options. This will address the situation where someone is seeking an explanation for not seeing the CA or Verification field.

kilfoyle replied:
For `Server SSL certificate authorities` I updated the text to:
> This setting is optional. This setting is not available when the authentication `None` and `Plaintext` options are selected.

And for `Verification mode`: 
> The default value is `Full`. This setting is not available when the authentication `None` and `Plaintext` options are selected.

szwarckonrad commented on docs/en/ingest-management/fleet/fleet-settings-output-kafka.asciidoc:
```diff
@@ -0,0 +1,367 @@
+Then, create a processor to route critical events:
+
+* Condition: `Contains`
+* Value: `message: “CRITICAL”`  
```
It could be valuable to emphasize that the value must adhere to the format of '[key]: value', where both the key and value are strings. For instance: host.port: 2000, message: Hello

szwarckonrad commented:
Looks great! I've added a few comments - just some suggestions on how we might enhance clarity.

Another consideration arose today - whenever an user selects the Kafka output type, we will display the warning detailed in [this ticket](https://github.com/elastic/security-team/issues/7309) and depicted in [this pull request](https://github.com/elastic/kibana/pull/163579).

kilfoyle commented:
Thanks for the excellent feedback!

@szwarckonrad I took all your suggestions and added a tip at the top of the docs page `Kafka output is currently not supported on {agents} using the {elastic-defend} integration.`.

 @juliaElastic Good idea to add a graphic. I've added one to help people find the `Add output` button. I didn't add the `Add new output` UI itself. We try to minimize showing larger UIs since they change frequently and the docs can go out of sync easily.
